pkg/agent/agent.go

```diff
@@ -423,19 +432,28 @@ func newHandlers() []handler.Handler {
 	handlers := []handler.Handler{
 		handler.NewAPISvcHandler(agent.cacheManager),
 		handler.NewInstanceHandler(agent.cacheManager),
-		handler.NewCategoryHandler(agent.cacheManager),
 		handler.NewAgentResourceHandler(agent.agentResourceManager),
-		handler.NewCRDHandler(agent.cacheManager),
-		handler.NewARDHandler(agent.cacheManager),
-		// handler.NewACLHandler(agent.cacheManager),
 		agent.proxyResourceHandler,
 	}
 
+	if agent.cfg.GetAgentType() == config.DiscoveryAgent {
```

Comment:
Only registering handlers based on what is defined in the watch topic for the agent type.